platform/include/transport_interface.h

```diff
@@ -19,21 +19,146 @@
+ * been made before invoking any library routines that receive data over the
+ * network.
+ * The @ref TransportRecv_t is typically implemented by calling directly the
+ * TLS layer function to receive data. If one is transferring over plaintext TCP,
+ * without TLS, then a call is made directly to the TCP layer function to receive
+ * data. @ref TransportRecv_t may be invoked multiple times by the library if
+ * fewer bytes than were requested to receive are returned.
+ * <br><br>
+ * <b>Example code:</b>
+ * @code{c}
+ * int32_t myNetworkRecvImplementation( const NetworkContext_t * pNetworkContext,
+ *                                      void * pBuffer,
+ *                                      size_t bytesToRecv )
+ * {
+ *     int32_t bytesReceived = 0;
+ *     bytesReceived = mySocketRecv( pNetworkContext->tcpSocketContext,
+ *                                   pBuffer,
+ *                                   bytesToRecv,
+ *                                   MY_SOCKET_TIMEOUT );
+ *     if( bytesReceived < 0 )
+ *     {
+ *         // Handle socket error.
+ *     }
+ *     // Handle other cases.
+ *
+ *     return bytesReceived;
+ * }
+ * @endcode
+ * <br>
+ * -# Implementing @ref TransportSend_t<br><br>
+ * @snippet this define_transportsend
+ * <br>
+ * This callback lets the application know that the library is ready to send
+ * data over the network. Connection with the server of interest should have
+ * been made before invoking any library routines that send data over the
+ * network.
+ * The @ref TransportSend_t is typically implemented by calling directly the
+ * TLS layer function to send data. If one is transferring over plaintext TCP,
+ * without TLS, then a call is made directly to the TCP layer function to send
+ * data. @ref TransportSend_t may be invoked multiple times by the library if
+ * fewer bytes than were requested to send are returned.
+ * <br><br>
+ * <b>Example code:</b>
+ * @code{c}
+ * int32_t myNetworkSendImplementation( const NetworkContext_t * pNetworkContext,
+ *                                      const void * pBuffer,
+ *                                      size_t bytesToSend )
+ * {
+ *     int32_t bytesSent = 0;
+ *     bytesSent = mySocketSend( pNetworkContext->tcpSocketContext,
+ *                               pBuffer,
+ *                               bytesToSend,
+ *                               MY_SOCKET_TIMEOUT );
+ *     if( bytesSent < 0 )
+ *     {
+ *         // Handle socket error.
+ *     }
+ *     // Handle other cases.
+ *
+ *     return bytesSent;
+ * }
+ * @endcode
+ */
+
+/**
+ * @transportstruct
+ * @typedef NetworkContext_t
  * @brief The NetworkContext is an incomplete type. An implementation of this
- * interface must define NetworkContext as per the requirements. This context
- * is passed into the network interface functions.
+ * interface must define struct NetworkContext for their system's requirements.
```

Comment:
their system's?  maybe "the system requirements"

Reply:
Sure thanks